### ISSUE TITLE
[BUG] `default.qubit.jax` incompatible with the parameter-shift rule

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#1574]
### Expected behavior

The `default.qubit.jax` device works well even when using the parameter-shfit rule.

### Actual behavior

For the tape interface, computing the jacobian of the tape provides incorrect results:

```python
import pennylane as qml
from pennylane import numpy as np
import jax

dev = qml.device('default.qubit.jax', wires=2)

params = jax.numpy.array(0.3)

def cost_fn(x):
    with qml.tape.JacobianTape() as tape1:
        qml.Hadamard(wires=[0])
        qml.RY(x, wires=[0])
        qml.CNOT(wires=[0, 1])
        qml.expval(qml.PauliZ(0))

    return tape1.jacobian(dev)

res = cost_fn(params)
print(res, type(res))
```
```
[[0.]] <class 'numpy.ndarray'>
```
For the QNode interface with `qml.grad`, the gradient is not being computed, execution hangs (likely the kernel dies):
```python
import pennylane as qml
from pennylane import numpy as np
import jax

dev = qml.device('default.qubit.jax', wires=2)
params = jax.numpy.array(0.3)

interface = 'jax'

@qml.qnode(dev, interface=interface, diff_method='parameter-shift')
def cost_fn(x):
    qml.Hadamard(wires=[0])
    qml.RY(x, wires=[0])
    qml.CNOT(wires=[0, 1])
    return qml.expval(qml.PauliZ(0))

res = jax.grad(cost_fn)(params)
res
```

### Additional information

Changing to `diff_method='backprop` or changing the device produces the correct result:
```
DeviceArray(-0.95533645, dtype=float32)
```

The issue for the parameter shift rule with the QNode is likely related to calling `host_callback.call` with an incorrect shape and/or data type being passed. This Jax function is experimental and can lead to the observed behaviour when incorrect arguments are being passed.

The https://github.com/PennyLaneAI/pennylane/issues/1412 issue might also be related to some extent.

As the `default.qubit.jax` device is mainly used with `diff_method='backprop'`, this issue likely doesn't affect users and is not high priority, but rather good to know about.

### System information

```shell
Name: PennyLane
Version: 0.17.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /xanadu/pennylane
Requires: numpy, scipy, networkx, autograd, toml, appdirs, semantic-version, autoray, cachetools
Required-by: PennyLane-Orquestra, PennyLane-Qchem, PennyLane-SF, pennylane-qulacs, PennyLane-qiskit, PennyLane-Cirq, PennyLane-IonQ, amazon-braket-pennylane-plugin, PennyLane-Forest, PennyLane-Lightning, PennyLane-Honeywell
Platform info:           Linux-5.11.0-27-generic-x86_64-with-glibc2.10
Python version:          3.8.5
Numpy version:           1.21.1
Scipy version:           1.7.1
Installed devices:
- orquestra.forest (PennyLane-Orquestra-0.15.0)
- orquestra.ibmq (PennyLane-Orquestra-0.15.0)
- orquestra.qiskit (PennyLane-Orquestra-0.15.0)
- orquestra.qulacs (PennyLane-Orquestra-0.15.0)
- strawberryfields.fock (PennyLane-SF-0.16.0.dev0)
- strawberryfields.gaussian (PennyLane-SF-0.16.0.dev0)
- strawberryfields.gbs (PennyLane-SF-0.16.0.dev0)
- strawberryfields.remote (PennyLane-SF-0.16.0.dev0)
- strawberryfields.tf (PennyLane-SF-0.16.0.dev0)
- qulacs.simulator (pennylane-qulacs-0.17.0.dev0)
- qiskit.aer (PennyLane-qiskit-0.16.0)
- qiskit.basicaer (PennyLane-qiskit-0.16.0)
- qiskit.ibmq (PennyLane-qiskit-0.16.0)
- cirq.mixedsimulator (PennyLane-Cirq-0.17.0.dev0)
- cirq.pasqal (PennyLane-Cirq-0.17.0.dev0)
- cirq.qsim (PennyLane-Cirq-0.17.0.dev0)
- cirq.qsimh (PennyLane-Cirq-0.17.0.dev0)
- cirq.simulator (PennyLane-Cirq-0.17.0.dev0)
- ionq.qpu (PennyLane-IonQ-0.17.0.dev0)
- ionq.simulator (PennyLane-IonQ-0.17.0.dev0)
- braket.aws.qubit (amazon-braket-pennylane-plugin-1.4.1.dev0)
- braket.local.qubit (amazon-braket-pennylane-plugin-1.4.1.dev0)
- forest.numpy_wavefunction (PennyLane-Forest-0.17.0.dev0)
- forest.qvm (PennyLane-Forest-0.17.0.dev0)
- forest.wavefunction (PennyLane-Forest-0.17.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.16.0.dev0)
- honeywell.hqs (PennyLane-Honeywell-0.16.0.dev0)
- default.gaussian (PennyLane-0.17.0)
- default.mixed (PennyLane-0.17.0)
- default.qubit (PennyLane-0.17.0)
- default.qubit.autograd (PennyLane-0.17.0)
- default.qubit.jax (PennyLane-0.17.0)
- default.qubit.tf (PennyLane-0.17.0)
- default.tensor (PennyLane-0.17.0)
- default.tensor.tf (PennyLane-0.17.0)
```
```


### 

- [X] I have searched exisisting GitHub issues to make sure the issue does not already exist.

**Comment by albi3ro** [comment]
I'll take a look into this today,

**Comment by antalszava** [comment]
@albi3ro maybe it's good to have on our radar, but perhaps not the highest of priority. Unless the user explicitly specifies `default.qubit.jax` with parameter shift, this should not be an issue.

**Comment by josh146** [comment]
Agree, happy to leave this low-priority for now, since it's unlikely that `default.qubit.jax` and `parameter-shift` will be created automatically (you'd have to specify this manually to get it)

**Comment by antalszava** [comment]
This now works well on `master` (likely due to the new batch execution pipeline) with `jax==0.2.24` and `jaxlib==0.1.73`.
